include/rocksdb/compaction_service.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019-present, Rockset, Inc.  All rights reserved.
+// This source code is licensed under both the GPLv2 (found in the
+// COPYING file in the root directory) and Apache 2.0 License
+// (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <stdint.h>
+
+#include <climits>
+#include <vector>
+
+#include "rocksdb/customizable.h"
+#include "rocksdb/options.h"
+#include "rocksdb/types.h"
+
+// An EXPERIMENTAL feature to allow alternative CompactionService
+// policies to be implemented.  A future derivative of this interface
+// could be used, for example, to offload a compaction job to a remote
+// process.
+//
+// This feature is an EXPERIMENTAL work in progress and is subject to
+// change in a future release
+namespace ROCKSDB_NAMESPACE {
+class ColumnFamilyHandle;
+class Slice;
+
+struct CompactionJobInfo;
+struct ConfigOptions;
+struct FileOptions;
+
+// The input parameters for a compaction service request.
+// The fields in this structure are EXPERIMENTAL and subject to change as the
+// interface is refined.
+struct CompactionServiceOptions : CompactionOptions {
+ public:
+  // The identity of the database being compacted
+  std::string db_id;
+
+  // The handle of the column family to compact
+  ColumnFamilyHandle *column_family;
+
+  // The DBOptions for this compaction.
+  DBOptions db_options;
+
+  // List of existing snapshots in the db
+  std::vector<SequenceNumber> existing_snapshots;
+
+  // List of input files of the compaction
+  std::vector<std::string> input_files;
```

Comment:
We might still need a blob (a binary format) parameter for compaction details. For example, manifest file needed for this compaction, for the first version, we plan to load manifest file from the remote disk. I think we could expand that later.